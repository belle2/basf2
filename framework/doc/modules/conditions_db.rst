.. automodule:: conditions_db
   :members:

.. automodule:: conditions_db.iov
   :members:
