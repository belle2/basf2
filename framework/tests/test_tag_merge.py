##########################################################################
# basf2 (Belle II Analysis Software Framework)                           #
# Author: The Belle II Collaboration                                     #
#                                                                        #
# See git log for contributors and copyright holders.                    #
# This file is licensed under LGPL-3.0, see LICENSE.md.                  #
##########################################################################

'''
Test for checking if b2conditionsdb-tag-merge behaves as expted:
- highest priority to the first tag in the list;
- within the same tag, highest priority to the payloads with the highest revision number.
'''


import subprocess

import conditions_db as cdb


if __name__ == '__main__':

    tags = ['main_tag_merge_test_1', 'main_tag_merge_test_2', 'main_tag_merge_test_3']
    final_tag = 'main_tag_merge_test_0'

    db = cdb.ConditionsDB()

    for tag in tags:
        payloads = db.get_all_iovs(globalTag=tag)
        payloads.sort(key=lambda p: p.iov)

        for payload in payloads:
            print(f'  {payload.name} r{payload.revision} {payload.iov}')

    subprocess.check_call(['b2conditionsdb-tag-merge', '-o', final_tag, '--dry-run'] + tags)

    subprocess.check_call(['b2conditionsdb-tag-merge', '-o', final_tag, '--dry-run', '--run-range', '5', '0', '5', '300'] + tags)
