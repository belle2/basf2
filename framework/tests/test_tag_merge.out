[INFO] Steering file: test_tag_merge.py
[INFO] Obtaining list of iovs for globaltag main_tag_merge_test_1
  KLMStripEfficiency r2587 (5, 0, 5, 9)
  KLMStripEfficiency r2553 (5, 0, 5, 200)
  KLMStripEfficiency r2615 (5, 10, 5, 35)
  KLMStripEfficiency r2670 (5, 36, 5, 87)
  KLMStripEfficiency r2704 (5, 87, 5, 90)
  KLMStripEfficiency r2810 (5, 91, 5, 100)
  KLMStripEfficiency r2553 (5, 300, 5, 999)
  KLMStripEfficiency r2553 (5, 1100, 5, -1)
[INFO] Obtaining list of iovs for globaltag main_tag_merge_test_2
  KLMStripEfficiency r2670 (5, 0, 5, 100)
  KLMStripEfficiency r2914 (5, 199, 5, 250)
  KLMStripEfficiency r2999 (5, 251, 5, 298)
  KLMStripEfficiency r2670 (5, 299, 5, 299)
  KLMStripEfficiency r3001 (5, 300, 5, 399)
  KLMStripEfficiency r3112 (5, 400, 5, 499)
[INFO] Obtaining list of iovs for globaltag main_tag_merge_test_3
  KLMStripEfficiency r3001 (5, 0, 5, 999)
  KLMStripEfficiency r3205 (5, 999, 5, 1050)
  KLMStripEfficiency r3299 (5, 1051, 5, -1)
[INFO] Obtaining list of iovs for globaltag main_tag_merge_test_1
[ERROR] Overlap for payload KLMStripEfficiency r2670: Overlap between (5, 87, 5, 100) and (5, 36, 5, 87)
[ERROR] Overlap for payload KLMStripEfficiency r2553: Overlap between (5, 0, 5, 35) and (5, 0, 5, 200)
[INFO] Obtaining list of iovs for globaltag main_tag_merge_test_2
[INFO] Obtaining list of iovs for globaltag main_tag_merge_test_3
[ERROR] Overlap for payload KLMStripEfficiency r3001: Overlap between (5, 999, 5, inf) and (5, 0, 5, 999)
[INFO] Result of merging the globaltags main_tag_merge_test_1, main_tag_merge_test_2, main_tag_merge_test_3
--------------------------------------------------------------------------------
Name                      Rev  First  First  Final  Final  Source
                               Exp    Run    Exp    Run    
--------------------------------------------------------------------------------
KLMStripEfficiency        2587 5      0      5      9      main_tag_merge_test_1
KLMStripEfficiency        2615 5      10     5      35     main_tag_merge_test_1
KLMStripEfficiency        2670 5      36     5      86     main_tag_merge_test_1
KLMStripEfficiency        2704 5      87     5      90     main_tag_merge_test_1
KLMStripEfficiency        2810 5      91     5      100    main_tag_merge_test_1
KLMStripEfficiency        2553 5      101    5      200    main_tag_merge_test_1
KLMStripEfficiency        2914 5      201    5      250    main_tag_merge_test_2
KLMStripEfficiency        2999 5      251    5      298    main_tag_merge_test_2
KLMStripEfficiency        2670 5      299    5      299    main_tag_merge_test_2
KLMStripEfficiency        2553 5      300    5      999    main_tag_merge_test_1
KLMStripEfficiency        3205 5      1000   5      1050   main_tag_merge_test_3
KLMStripEfficiency        3299 5      1051   5      1099   main_tag_merge_test_3
KLMStripEfficiency        2553 5      1100   5      -1     main_tag_merge_test_1
[INFO] Obtaining list of iovs for globaltag main_tag_merge_test_1
[ERROR] Overlap for payload KLMStripEfficiency r2670: Overlap between (5, 87, 5, 100) and (5, 36, 5, 87)
[ERROR] Overlap for payload KLMStripEfficiency r2553: Overlap between (5, 0, 5, 35) and (5, 0, 5, 200)
[INFO] Obtaining list of iovs for globaltag main_tag_merge_test_2
[INFO] Obtaining list of iovs for globaltag main_tag_merge_test_3
[INFO] Result of merging the globaltags main_tag_merge_test_1, main_tag_merge_test_2, main_tag_merge_test_3
--------------------------------------------------------------------------------
Name                      Rev  First  First  Final  Final  Source
                               Exp    Run    Exp    Run    
--------------------------------------------------------------------------------
KLMStripEfficiency        2587 5      0      5      9      main_tag_merge_test_1
KLMStripEfficiency        2615 5      10     5      35     main_tag_merge_test_1
KLMStripEfficiency        2670 5      36     5      86     main_tag_merge_test_1
KLMStripEfficiency        2704 5      87     5      90     main_tag_merge_test_1
KLMStripEfficiency        2810 5      91     5      100    main_tag_merge_test_1
KLMStripEfficiency        2553 5      101    5      200    main_tag_merge_test_1
KLMStripEfficiency        2914 5      201    5      250    main_tag_merge_test_2
KLMStripEfficiency        2999 5      251    5      298    main_tag_merge_test_2
KLMStripEfficiency        2670 5      299    5      299    main_tag_merge_test_2
KLMStripEfficiency        2553 5      300    5      300    main_tag_merge_test_1
